### DISCUSSION TITLE
srsRAN 4G 23.04

### DISCUSSION THREAD

**Discussion by andrepuschmann** [srsran/srsRAN_4G#1158] [Announcements]
* Introduced configurable s1 connection timer
* Updated 4G RRC ASN.1 to Rel 17
* Added reestablishment support during S1-Handover
* Added basic support for NSSAI based slicing in UE & gNodeB
* Updated the RRC to enable srsUE compatibility with new srsgnb
* Updated eMBMS to fix various outstanding issues
* Added basic support for RIC E2 interface
* other bug-fixes and improved stability and performance in all parts

<hr /><em>This discussion was created from the release <a href='https://github.com/srsran/srsRAN_4G/releases/tag/release_23_04'>srsRAN 4G 23.04</a>.</em>

**Comment by mbalalzade** [comment]
hello. please accept my excuse for getting your time. i am new to srsran i am try to lean the code but its too hard for me. in this time i am working on souding reference signal can u help me what are the steps i should do to understand the structure and concepts of srsran coding . best wishes. thanks

**Reply by jjose001** [reply]
I am in the same situation, if you have notices, reply it! 
Thank you

**Comment by mbalalzade** [comment]
Hello. In 6 month ago i started to reading the codes. Maybee i do not know about many c and cpp concepts but when i try to play with code for example every function input and output coming from which function using printf. But notice u can be export in one feild after some month because its a little hard to find relation between each part. Also never forget to use sharetechnote site and a site related to daniel esteves. Goodluck
    On Tuesday, November 12, 2024 at 04:07:34 PM GMT+3:30, jjose001 ***@***.***> wrote:  
 
 


I am in the same situation, if you have notices, reply it!
Thank you

**Reply by jjose001** [reply]
Hi @mbalalzade, 
To understand the code is a difficult work that I have to made if i want to manipulate srsRAN as i want. Im going to continue analising.

Thank you for your contribution.
